### DISCUSSION TITLE
It is safe to use private css variables declared in mantine components?

### DISCUSSION THREAD

**Discussion by 7iomka** [mantinedev/mantine#4870] [Q&A] [answered]
I want for input to use resolved value of padding left
I seen you have the private variable
```
  --_input-left-section-size: var(--input-left-section-width, calc(var(--_input-height) - 2px))
```

so when user don't pass --input-left-section-width directly, it is resolved my input height that is resolved by passed size (with fallback).

If we not able to use this css vars, we need to replicate all this logic by hand.
So, question is - it is safe to use it now?

or are we assuming that the internal api of variables may change dramatically in the foreseeable future?

**Comment by rtivital** [comment, marked as answer]
I would not advise using it, this API is private and will not be mentioned in the changelogs. You can safely use variables without `_` underscore, they are considered public.
